lib/fs.js

```diff
@@ -175,20 +176,15 @@ function access(path, mode, callback) {
   }
 
   path = getValidatedPath(path);
-
-  mode = mode | 0;
+  mode = getValidMode(mode, 7);
   const req = new FSReqCallback();
   req.oncomplete = makeCallback(callback);
   binding.access(pathModule.toNamespacedPath(path), mode, req);
 }
 
 function accessSync(path, mode) {
   path = getValidatedPath(path);
-
-  if (mode === undefined)
-    mode = F_OK;
-  else
-    mode = mode | 0;
+  mode = getValidMode(mode, 7);
```

Comment:
This makes too many assumptions about the value of `R_OK|W_OK|X_OK` and also seems to assume that `F_OK == 0` (which it probably is but that's not guaranteed.)

Also applies to line 179.

Reply:
I fixed that by returning `F_OK` in case mode is set to `undefined` and by checking for the actual value of `R_OK|W_OK|X_OK|F_OK`.